app/src/test/java/com/antonioleiva/mvpexample/app/main/MainPresenterTest.java

```diff
@@ -0,0 +1,76 @@
+package com.antonioleiva.mvpexample.app.main;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertNull;
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MainPresenterTest {
+
+    @Mock
+    MainView view;
+    @Mock
+    FindItemsInteractor interactor;
+
+    MainPresenterImpl presenter;
+
+    @Before
+    public void setUp() throws Exception {
+        presenter = new MainPresenterImpl(view);
+        Field field = presenter.getClass().getDeclaredField("findItemsInteractor");
+        field.setAccessible(true);
+        field.set(presenter, interactor);
+    }
+
+    @Test
+    public void checkIfShowsProgressOnResume() {
+        presenter.onResume();
+        verify(view, times(1)).showProgress();
+    }
+
+    @Test
+    public void checkIfShowsMessageOnItemClick() {
+        presenter.onItemClicked(1);
+        verify(view, times(1)).showMessage(anyString());
+    }
+
+    @Test
+    public void checkIfRightMessageIsDisplayed() {
+        ArgumentCaptor<String> captor = forClass(String.class);
+        presenter.onItemClicked(1);
+        verify(view, times(1)).showMessage(captor.capture());
+        assertThat(captor.getValue(), is("Position 2 clicked"));
+    }
+
+    @Test
+    public void checkIfViewIsReleasedOnDestroy() throws Exception {
+        presenter.onDestroy();
+        Field field = presenter.getClass().getDeclaredField("mainView");
```

Comment:
I'm not very convinced about reflection here. I prefer you to change the base code so that it allows testing in a more natural way.